src/data-access/GroupRepository.js

```diff
@@ -0,0 +1,52 @@
+import database from '../models/index';
+
+export const saveGroup = async (newGroup) => {
+    return await database.Group.create(newGroup);
+};
+
+export const getGroup = async (id) => {
+    return await database.Group.findOne({ where: { id } });
+};
+
+export const getAll = async () => {
+    return await database.Group.findAll();
+};
+
+export const updateGroup = async (id, group) => {
+    const groupToUpdate = await database.Group.findOne({ where: { id } });
+    if (groupToUpdate) {
+        const result = await database.Group.update(group, { where: { id } });
+        if (result > 0) {
+            return group;
+        }
+    }
+    return null;
+};
+
+export const removeGroup = async (id) => {
+    const toRemove = await database.Group.findOne({ where: { id } });
+    if (toRemove) {
+        const result = await database.Group.destroy({ where: { id } });
+        if (result > 0) {
+            return toRemove;
+        }
+    }
+    return null;
+};
+
+export const saveUserGroup = async (groupId, userIds) => {
+    const group = await database.Group.findOne({ where: { id : groupId } });
+    if (group) {
+        const users = await database.User.findAll({ where: { id : userIds, isDeleted : false } });
+        if (users && users.length === userIds.length) {
+            await database.sequelize.transaction(async (t) => {
+                await group.addUsers(userIds, { transaction: t });
+            });
+            return await group.getUsers({ through: { group_id: groupId } });
+        }
+        const userIdsFromDB = users.map((u) => (u.id));
+        const notFoundUserIds = userIds.filter((id) => !userIdsFromDB.includes(id));
+        return { status: 'Not Found', message : `users with ids: ${JSON.stringify(notFoundUserIds)} aren't found` };
+    }
+    return { status: 'Not Found', message : `group with id: ${groupId} isn't found` };
```

Comment:
You constructing error response inside on repository. In this case your internal/low-level components aware about external/upper components. In this case your architecture became tightly coupled and in is harder to extend.

From my point it is better to throw exception here with some message/reason or return null and only in controller you will construct error message based on controller logic

Reply:
Yes, I moved status to controller instead.

Reply:
Problem not with status filed/attribute but with whole object. For example when you call `removeGroup` then you return null in case group was not found. in case of `saveUserGroup` you return object with message. For me it is strange.
I think that would be better to return null in this case and based on this null value controller can decide what kind of response should be send to the client